doc/classes/RenderingServer.xml

```diff
@@ -2639,6 +2639,9 @@
 				  - Position + Custom data: 16 floats (12 floats for Transform3D, 4 floats of custom data)
 				  - Position + Vertex color + Custom data: 20 floats (12 floats for Transform3D, 4 floats for Color, 4 floats of custom data)
 				[/codeblock]
+				Instance transforms are in row-major order. Specifically:
+				- For [Transform2D] the float-order is: [code](x.x, y.x, padding_float, origin.x, x.y, y.y, padding_float, origin.y)[/code].
+				- For [Transform3D] the float-order is: [code](basis.x.x, basis.y.x, basis.z.x, origin.x, basis.x.y, basis.y.y, basis.z.y, origin.y, basis.x.z, basis.y.z, basis.z.z, origin.z)[/code].
```

Comment:
I think this slightly out of order. 

Here is the order used:
https://github.com/godotengine/godot/blob/4254946de93bab0cc1fb36a7b9039c9ec43be924/servers/rendering/renderer_rd/storage_rd/mesh_storage.cpp#L1728-L1739

```suggestion
				- For [Transform3D] the float-order is: [code](basis.x.x, basis.x.y, basis.x.z, origin.x, basis.y.x, basis.y.y, basis.y.z, origin.y, basis.z.x, basis.z.y, basis.z.z, origin.z)[/code].
```

Reply:
Yeah so the order is correct as is, `basis.x`/`basis.y`/`basis.z` refer to columns, not to rows.

https://github.com/godotengine/godot/blob/4254946de93bab0cc1fb36a7b9039c9ec43be924/doc/classes/Basis.xml#L377-L388

https://github.com/godotengine/godot/blob/4254946de93bab0cc1fb36a7b9039c9ec43be924/core/math/basis.h#L173-L183

<details><summary>Relevant source registering `Basis::get_column`/`Basis::set_column` as accessors </summary>
<p>

https://github.com/godotengine/godot/blob/4254946de93bab0cc1fb36a7b9039c9ec43be924/core/variant/variant_setget.cpp#L119-L121
https://github.com/godotengine/godot/blob/4254946de93bab0cc1fb36a7b9039c9ec43be924/core/variant/variant_setget.cpp#L66
https://github.com/godotengine/godot/blob/4254946de93bab0cc1fb36a7b9039c9ec43be924/core/variant/variant_setget.h#L320-L322
https://github.com/godotengine/godot/blob/4254946de93bab0cc1fb36a7b9039c9ec43be924/core/variant/variant_setget.h#L238-L267

</p>
</details> 

---

Similarily `Basis[int index]` operator in core/C++ operates on rows but in GDScript/C# it operates on columns. Yeah, this caused some confusion/bugs :upside_down_face:; but currently it's much cleaner than it used to be (cleaner naming in code like `rows` instead of `elements`, better documentation, etc.).

Reply:
Thank you for that explanation. Indeed I was working off my knowledge of using it from C++ so I had it backwards